sc2ts/info.py

```diff
@@ -235,6 +235,48 @@ def get_path_mrca(path1, path2, node_time):
     )
 
 
+@numba.njit
+def _get_num_muts(
+    tree_nodes_preorder,
+    tree_parent_array,
+    tree_nodes_num_mutations,
+):
+    num_muts = np.zeros(len(tree_nodes_preorder), dtype=np.int32)
+    for node in tree_nodes_preorder:
+        pa = tree_parent_array[node]
+        if pa == -1:
+            num_muts[node] = 0  # Root
+        else:
+            assert num_muts[pa] >= 0
+            num_muts[node] = num_muts[pa]
+            num_muts[node] += tree_nodes_num_mutations[node]
+    return num_muts
+
+
+def get_num_muts(ts):
+    # All nodes msut be present in all trees.
+    num_muts_all_trees = np.zeros(ts.num_nodes, dtype=np.int32)
+    i = 0
+    for tree in ts.trees():
+        tree_nodes_preorder = tree.preorder()
+        assert ts.num_nodes == len(tree_nodes_preorder)
+        tree_site_ids = np.array([s.id for s in tree.sites()], dtype=np.int32)
```

Comment:
mutations_position = ts.sites_position[ts.mutations_site]
tree_mutations = (tree.interval.left <= mutations_position) & (mutations_position < tree.interval.right)

bincount(ta.mutations_node[tree_mutations],...)

Reply:
Iterating over sites is slow, and np.in is slower than a simple Boolean mask